### DISCUSSION TITLE
Outline Parse Error

### DISCUSSION THREAD

**Discussion by xiaohuihuitop** [microsoft/vscode-cpptools#9869] [Q&A]
![outline](https://user-images.githubusercontent.com/64055377/189417430-afd8a433-2f59-4ee7-9ffe-804f4ca8bc6a.png)

#if 0  
Data should not be displayed
#else
Data Should be displayed
#endif

But now the processing is not correct。
There is no correct identification condition judgment !

**Comment by bobbrow** [comment]
The Outline mode does not take into account the preprocessor conditions.  In other words, this is the expected behavior.  It will allow you to navigate quickly to all symbols in the file.  The editor will display the dimmed text as your screenshot shows and will provide IntelliSense for the sections that will be compiled.

**Reply by xiaohuihuitop** [reply]
Whether the preprocessing condition can be supported subsequently. Provide a setting for the user.

**Reply by sean-mcmanus** [reply]
@xiaohuihuitop Did you want to create an issue instead of a discussion?
